tests/assemblers/test_xgboost.py

```diff
@@ -247,23 +247,25 @@ def test_regression_saved_without_feature_names():
                     ast.FeatureRef(12),
                     ast.NumVal(9.72500038),
                     ast.CompOpType.GTE),
-                ast.NumVal(1.6614188),
-                ast.NumVal(2.91697121)),
+                ast.NumVal(4.98425627),
+                ast.NumVal(8.75091362)),
             ast.BinNumOpType.ADD),
         ast.IfExpr(
             ast.CompExpr(
                 ast.FeatureRef(5),
                 ast.NumVal(6.94099998),
                 ast.CompOpType.GTE),
-            ast.NumVal(3.33810854),
-            ast.NumVal(1.71813202)),
+            ast.NumVal(8.34557438),
+            ast.NumVal(3.9141891)),
         ast.BinNumOpType.ADD)
 
     assert utils.cmp_exprs(actual, expected)
 
 
 def test_linear_model():
+    # Default updater ("shotgun") is nondeterministic
```

Comment:
Refer to
> updater [default= shotgun]
       Choice of algorithm to fit linear model
           shotgun: Parallel coordinate descent algorithm based on shotgun algorithm. Uses ‘hogwild’ parallelism and therefore produces a nondeterministic solution on each run.
https://xgboost.readthedocs.io/en/latest/parameter.html?highlight=shotgun#parameters-for-linear-booster-booster-gblinear

and
> Using gblinear booster with shotgun updater is nondeterministic as
           it uses Hogwild algorithm.
https://github.com/dmlc/xgboost/blob/cd7d6f7d596abfe8eb81651386d8c05507c28fc8/python-package/xgboost/sklearn.py#L105-L106

It causes different results in API test which leads to impossibility to write `expected` expression.